enhancements/plain-k8s-bundles.md

```diff
@@ -0,0 +1,240 @@
+
+Security impacts are minimal because the number of kubernetes objects that OLM supports is still the same - just the delivery mechanism (the bundle format) is changing.
+
+## Design Details
+
+### Test Plan
+
+There will be e2e tests alongside unit tests. The e2e test will verify that the functionality is delivered as expected - bundles consisting just of kubernetes manifests are installed and managed on cluster by OLM. Potentially existing tests can be reused but with the newer bundle format.
+
+Outside of core OLM there will also be tests in operator registry to ensure the new bundle format is consistently generated as we expect. Its also critical to test how indexes composed of k8s+v1 bundles will behave.
+
+Additional tests will ideally include the console as well to ensure all the components are supporting the new functionality as expected.
+
+### Upgrade / Downgrade Strategy
+  
+Upgrade (not supporting `k8s+v1` mediatype -> supporting `k8s+v1` mediatype)
+
+- Older versions of OpenShift may install newer catalog source images that contain both the new `k8s+v1` format and the legacy `registry+v1` format. Its important that OLM continues to behave as expected in these cases, even if it cannot install the `k8s+v1` bundles explicitly. This may require a small patch to be backported to 4.x versions of OpenShift. This would not necessarily affect upstream OLM users as they can simply update OLM to use the new feature.
+- An updated version of OLM that supports the `k8s+v1` mediatype should be a safe upgrade - if the catalogsources do not have any bundles using the new format, there should be no change. Bundles in the `k8s+v1` mediatype format available in the updated cluster should be available to list via the packageserver and install.
+
+Downgrade (supporting `k8s+v1` mediatype -> not supporting `k8s+v1` mediatype)
+
+- OpenShift supports a downgrade path and so OLM must ensure that it can continue working even if the ability to install `k8s+v1` format is removed as OLM is downgraded to an older version. OLM should be able to reject bundle formats that it cannot successfully reconcile and emit a helpful error message in that case.
+
+### Version Skew Strategy
+
+How will the component handle version skew with other components?
+What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+
+- During an upgrade, we will always have skew among components, how will this impact your work?
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI, CRI
+  or CNI may require updating that component before the kubelet.
+
+## Implementation History
```

Comment:
Still reading, but it seems like from here down this is just the template. We should update this pr to remove these sections if they aren't valuable in this enhancement's context or update them to include the requested information.